openbb_terminal/dashboards/stream/forecast.py

```diff
@@ -0,0 +1,303 @@
+            x for x in st.session_state["df"] if x != "date"
+        ]
+        st.session_state["widget_options"]["column_widget"] = [
+            x for x in st.session_state["df"] if x != "date"
+        ]
+
+    def run(self):
+        r0c1 = st.columns([4, 1])
+        r1c1, r1c2, r1c3, r1c4, r1c5 = st.columns([2, 1, 1, 1, 1])
+        r2c1, r2c2, r2c3 = st.columns([1, 1, 1])
+
+        with r0c1:
+            st.title("Forecast")
+        with r1c1:
+            ticker = st.text_input(
+                "Ticker", "", key="ticker", on_change=self.on_ticker_change
+            )
+        with r1c2:
+            start_date = st.date_input(
+                "Start date", date.today() - timedelta(weeks=104), key="start"
+            )
+        with r1c3:
+            end_date = st.date_input("End date", date.today(), key="end")
+        with r1c4:
+            target_interval = st.selectbox(
+                "Interval", index=0, key="interval", options=interval_opts
+            )
+        with r1c5:
+            n_predict = st.selectbox(
+                "Prediction Days", index=3, key="n_predict", options=list(range(2, 31))
+            )
+        with r2c1:
+            # TODO: disable this if the current model does not allow for it
+            past_covs_widget = st.multiselect(
+                "Past Covariates",
+                options=st.session_state["widget_options"]["past_covs_widget"],
+            )
+        with r2c2:
+            target_widget = st.selectbox(
+                "Target", options=st.session_state["widget_options"]["target_widget"]
+            )
+        with r2c3:
+            model_widget = st.selectbox("Model", options=list(model_opts))
+
+        st.markdown("""---""")
+        if st.button("Get forecast"):
+            if ticker:
+                self.handle_changes(
+                    past_covariates=past_covs_widget,
+                    start=start_date,
+                    end=end_date,
+                    interval=target_interval,
+                    tickers=ticker,
+                    target_column=target_widget,
+                    model=model_widget,
+                    n_predict=n_predict,
+                    naive=False,
+                    forecast_only=False,
+                )
+            else:
+                st.write("Please select a ticker")
+
+            # TODO need to new add in business days and merge to make new DF before plotting
+            # create df with 5 next business days from a start date
+            # next_business_days = pd.date_range(start=end_date, periods=5, freq="B")
+
+        # if st.button("Add Column"):
+        # kwargs = {}
+        # if has_parameter(handler.feature_model, "target_column"):
+        # kwargs["target_column"] = handler.feature_target
+        # handler.df = handler.feature_model(handler.df, **kwargs)
+        # st.session_state["widget_options"]["past_covs_widget"] = handler.df.columns
```

Comment:
@colin99d should be here?

Reply:
Yeah. I will turn that into working code later.